src/helperFunctions.js

```diff
@@ -131,6 +134,48 @@ class helpers {
       })
   }
 
+  /**
+   *
+   * @param {string} url
+   * @param {number} ttl
+   * @param {string} httpMethod
+   *
+   * @returns {Promise}
+   */
+  async signUrl (url, ttl = 1200, httpMethod = 'get') {
+    const key = await this.getSignKey()
+    const user = await this.getCurrentUser()
+    const signUrl = new SignUrl({
+      credential: user.id,
+      secretKey: key,
+      ttl: ttl,
+      algorithm: 'sha512'
+    })
+    try {
+      const signed = signUrl.generateSignedUrl(url, httpMethod)
+      return Promise.resolve(signed)
```

Comment:
you might not even need to resolve or reject, that's the magic of async.

if you return a value it will be considered a resolution

and if you throw an exception, it's like rejecting

best to double check though